src/core/utils/string.cpp

```diff
@@ -106,6 +106,77 @@ bool Utils::String::naturalSort(QString left, QString right, bool &result)
     return false;
 }
 
+bool Utils::String::naturalLessThan(const QString &left, const QString &right)
+{
+    // Return value `false` indicates QString `right` should go before QString `left`, otherwise, after
+    const bool case_sensitive = false;
+    int posL = 0;
+    int posR = 0;
+    while (true) {
+        while (true) {
+            if (posL == left.size() || posR == right.size())
+                return (left.size() < right.size());  // when a shorter string is another string's prefix, shorter string place before longer string
+
+            QChar leftChar = left[posL];
+            QChar rightChar = right[posR];
+            if (leftChar.isDigit() && rightChar.isDigit())
+                break; // Both are digits, break this loop and compare numbers
+            else if (leftChar.isLetter() && rightChar.isLetter()) {
+                if(!case_sensitive) {
+                    // special case for case insensitive comparsion
+                    QChar l = leftChar.toLower();
+                    QChar r = rightChar.toLower();
+                    if(l != r)
+                        return (l < r);
+                }
+
+                // case sensitive comparsion
+                if (leftChar != rightChar) {
```

Comment:
>After this test why don't you always return with (leftChar < rightChar)?

natural sort tends to put lowercase letters before uppercase letters, hence the following `if..else..` section.